packages/vulcan-forms/lib/components/Form.jsx

```diff
@@ -450,7 +466,45 @@ class Form extends Component {
       return newState;
     });
   };
-
+  
+  handleRouteLeave = () => {
```

Comment:
I would argue behavior should be optional, toggled by a flag on the props, and default of being off. I love the idea, but it would need to be optional for sure for instance in the case where people use their routes for information:

/Page
/Page#value
/Page/value

Reply:
Hi Justin! I don't have a problem adding a prop to SmartForm to toggle this feature or even a global settings file value. Let's see what @SachaG  thinks. But remember this behaviour wouldn't show up on all pages, just pages with a SmartForm on it. Do you have a use case where the route changes but the user stays on the same SmartForm? Maybe we can find a way for handleRouteLeave to detect this.

Reply:
In general you don't want to have a default behavior for all forms to prompt a user about leaving a route. The decision should be on the developer to manually opt-in to UX behavior that causes a jarring action (prompting). Making this the default would impact any app out there that uses SmartForm (which we don't know how many that is). For them to upgrade and then realize anywhere they're using forms they now have this new prompting behavior would have unintended consequences for developers.

Reply:
I agree, I wasn't disagreeing with you.

Reply:
Gotcha, the only use case I'd have would be where you'd scroll to different portions of the page or form based on floating UI buttons (up / down arrows for instance).

Reply:
Yeah, that makes sense. I'll test that and make sure the warning doesn't come up in that case. How do you add the hash? Do you do it? Something like `location.hash = newHash; browserHistory.replace(location);` ?

Reply:
Should be able to do `withRouter` and do `props.router.replace`

Reply:
But maybe browser history I’d asssume?

Reply:
Hi @justinr1234! I just checked and when only the hash portion of the route changes, the alert does not come up.